src/components/login/GoogleAuthButton.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Button } from 'antd';
+
+export default function GoogleAuthButton({
+  onGoogleSignIn,
+}: {
+  onGoogleSignIn: (credential: string) => void;
+}) {
+  const [gsiScriptLoaded, setGsiScriptLoaded] = useState(false);
+  const clientId = useMemo(() => process.env.REACT_APP_GOOGLE_CLIENT_ID, []);
+
+  const handleGoogleSignIn = useCallback(
+    (res: any) => {
+      if (!res.clientId || !res.credential) return;
+      onGoogleSignIn(res.credential);
+    },
+    [onGoogleSignIn],
+  );
+
+  useEffect(() => {
+    if (gsiScriptLoaded) return;
+
+    const initializeGsi = () => {
+      // Typescript will complain about window.google
+      // Add types to your `react-app-env.d.ts` or //@ts-ignore it.
+      if (!(window as any).google || gsiScriptLoaded) return;
+
+      setGsiScriptLoaded(true);
```

Comment:
It’s not recommended to have the same React state be both mutable and a dependency within the same useEffect, as this can potentially lead to an infinite loop.

If we just need to know whether the scrip been loaded can we just do a simple dom finder to see if script with 'https://accounts.google.com/gsi/client' exists?

Reply:
Yep, pushed up a change for this. Thank you!